### PR TITLE
Handling anynomous user login

### LINKED ISSUE #331: [FEATURE] allow anonymous user to create an account
### Description

In the profile section if user is logged anonymously we can add a link or button which will navigate the user to the login page if they want to create their profile. This can improve the overall user experience.

### Screenshots

_No response_

### Additional information

Here's what you need to do for this issue:
- [ ] prompt anonymous user to create an account by asking them to input their email. 
- [ ] use the current loginID to register the user in Mongodb. Replace the objectid default from Mongodb to the current loginID
- [ ] update the auth for the user, in auth Context
- [ ] use kinde.com api to add the user to kinde after the successful register

### 👀 Have you checked if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Contributing Guidelines?

- [X] I have read and understood the rules in the [Contributing Guidelines](https://github.com/Dun-sin/Whisper/blob/main/CONTRIBUTING.md)

### Are you willing to work on this issue ?

none

### PR DESCRIPTION
**Pull request by bishwobista**
# Fixes Issue

**My PR closes #331 **

# 👨‍💻 Changes proposed(I added a function to handle new user login, if the user is anonymous or completely a new user )

# ✔️ Check List (Check all the applicable boxes)

- [x] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title and description of the PR is clear and explains the approach.

##  Note to reviewers

# 📷 Screenshots

Screenshot 2023-10-01 at 3 42 07 PM
Screenshot 2023-10-01 at 3 42 50 PM

### CODE REVIEW

**Comment by Dun-sin** [comment]
did you test that this works?

**Comment by Dun-sin** [comment]
> # PR Compliance Checks
> 
> Thank you for your Pull Request! We have run several checks on this pull request in order to make sure it's suitable for merging into this project. The results are listed in the following section.
> ## Conventional Commit PR Title
> 
> In order to be considered for merging, the pull request title must match the specification in [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/). You can edit the title in order for this check to pass. Most often, our PR titles are something like one of these:
> 
>     * docs: correct typo in README
> 
>     * feat: implement dark mode"
> 
>     * fix: correct remove button behavior
> 
> 
> Linting Errors
> 
>     * Found type "null", must be one of "feat","fix","docs","style","refactor","perf","test","build","ci","chore","revert"
> 
>     * No subject found

@bishwobista 
Please fix this error, if you don't know how, try this:
- Read the error carefully or read the contributing file
- Checkout past merged Pull requests to see how they were done for it to be merged

> Note: you can edit your PR message, this PR won't be merged if the error isn't gone and you have a week max to fix it before the PR gets closed without being merged

File: client/src/pages/Profile.jsx [hunk 1]
```diff
@@ -80,6 +80,52 @@ const Profile = () => {
         getProfileData(email)
     }, [])
 
+    const [getemail, setEmail] = useState('')
```
Comment (Dun-sin) [inline client/src/pages/Profile.jsx:83]: follow the style of the project, all hooks are at the top

File: client/src/pages/Profile.jsx [hunk 2]
```diff
@@ -80,6 +80,52 @@ const Profile = () => {
...
+            // Handling new user login
+            const data = {
+              email: getemail,
+              loginId: id,
+            };
+            try {
+              const response = await api.post('/login', data);
+              console.log(response.data.message);
+              // Updating the user info in local storage with the new data
+              localStorage.setItem('userInfo', JSON.stringify(response.data.userInfo));
```
Comment (Dun-sin) [inline client/src/pages/Profile.jsx:121]: use the authContext/reducer

File: client/src/pages/Profile.jsx [hunk 3]
```diff
@@ -80,6 +80,52 @@ const Profile = () => {
...
+              loginId: id,
+            };
+            try {
+              const response = await api.post('/login', data);
+              console.log(response.data.message);
+              // Updating the login object in local storage with the new user ID
+              const newAuth = { ...auth, loginId: response.data.userId };
+              localStorage.setItem('auth', JSON.stringify(newAuth));
+              // Updating the user info in local storage with the new data
+              localStorage.setItem('userInfo', JSON.stringify(response.data.userInfo));
```
Comment (Dun-sin) [inline client/src/pages/Profile.jsx:107]: use the auth reducer/context

File: client/src/pages/Profile.jsx [hunk 4]
```diff
@@ -80,6 +80,52 @@ const Profile = () => {
...
+          // Checking if the user is an anonymous user
+          const auth = JSON.parse(localStorage.getItem('auth'));
+          if (auth.loginType === 'anonymous') {
+            // Handling anonymous user login 
+            const data = {
+              email: getemail,
+              loginId: id,
+            };
+            try {
+              const response = await api.post('/login', data);
```
Comment (Dun-sin) [inline client/src/pages/Profile.jsx:101]: this is wrong, you didn't follow what i said at all

**Review by Dun-sin — CHANGES_REQUESTED** [review]

File: client/src/pages/Profile.jsx (hunk 4 above)
Comment (bishwobista) [reply client/src/pages/Profile.jsx:101]: I didn't understand, I have realized that this is beyond my knowledge. You can remove me from this one.

Comment (Dun-sin) [reply client/src/pages/Profile.jsx:101]: yeah i figured, was just waiting for you to admit it, but no worries. Thanks for wanting to contribute, you can check out Simpler issues or use the app and raise an issue.

**Pull request closed without merging**